rpki-validator/src/test/java/net/ripe/rpki/validator3/background/ThrottledTest.java

```diff
@@ -29,47 +29,46 @@
  */
 package net.ripe.rpki.validator3.background;
 
-import lombok.Getter;
-import lombok.Setter;
-import net.ripe.rpki.validator3.domain.validation.CertificateTreeValidationService;
-import net.ripe.rpki.validator3.storage.data.TrustAnchor;
-import net.ripe.rpki.validator3.storage.data.validation.CertificateTreeValidationRun;
-import org.quartz.DisallowConcurrentExecution;
-import org.quartz.Job;
-import org.quartz.JobBuilder;
-import org.quartz.JobDetail;
-import org.quartz.JobExecutionContext;
-import org.quartz.JobExecutionException;
-import org.quartz.JobKey;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.junit.Test;
 
-@DisallowConcurrentExecution
-public class CertificateTreeValidationJob implements Job {
+import java.util.concurrent.atomic.AtomicInteger;
 
-    public static final String TRUST_ANCHOR_ID_KEY = "trustAnchorId";
+import static org.junit.jupiter.api.Assertions.*;
 
-    @Autowired
-    private CertificateTreeValidationService validationService;
+public class ThrottledTest {
+    @Test
+    public void testTriggerDoenstTriggerTooOften() throws InterruptedException {
+        AtomicInteger counter = new AtomicInteger(0);
+        final Throttled<String> throttled = new Throttled<>(10);
 
-    @Getter
-    @Setter
-    private long trustAnchorId;
+        throttled.trigger("x", counter::incrementAndGet);
+        waitALittleToAllowExecutorToProcessRunnables();
+        assertEquals(1, counter.get());
 
-    @Override
-    public void execute(JobExecutionContext context) throws JobExecutionException {
-        validationService.validate(trustAnchorId);
+        for (int i = 0; i < 10; i++) {
+            throttled.trigger("x", counter::incrementAndGet);
+        }
+        Thread.sleep(1100);
+        assertEquals(1, counter.get());
```

Comment:
If I understand correctly, the first of this 10 additional trigger will be delayed for around 10 secs, and the rest will be rejected. So this will also holds true immediately after.
```
        Thread.sleep(10000);
        assertEquals(2, counter.get());
```

Reply:
I fixed the test so that it also checks for the case of "immediately trigger after".